### DISCUSSION TITLE
Downselecting a modeling language

### DISCUSSION THREAD

**Discussion by capsulecorplab** [ODU-CGA-CubeSat/sealion-mission-architecture#40] [General]
In order to document the flight software architecture using a model-based approach, we need to down-select a modeling language to support tasks, s.a., #2, #3, #9, & #11. The following is a trade-study of available modeling languages that support architecture modeling

| Criteria | [SysML v1](https://sysml.org/sysml-specs/) | [SysML v2](https://github.com/Systems-Modeling/SysML-v2-Release) | [PlantUML](https://plantuml.com/) | [M30ML](https://github.com/Mach30/m30ml) |
| ------------------------------------------------------------------ | - | - | - | - |
| Extensible ontology language                                 | x | x | x | x |
| Supports both textual & graphical view generation |   | x |     | x |
| Lightweight textual syntax                                       |   | x | x  | x |
| Relatively minimal overhead with modern doctools |   |    | x | x |
| Supports execution semantics                                 |   | x |    |   |

**Comment by capsulecorplab** [comment]
Leaning toward M30ML for architecture modeling, as it is currently the path of least resistance. This will also allow for easier integration with packaging the flight software as a [dof Component](https://github.com/Mach30/dof/blob/main/dist/DOF.adoc#component) for modeling CubeSat components, interfaces, & functions - since dof already includes a data structure definition for a `Component`, with plans to include `Interface` & `Function`

**Reply by Robb1994** [reply]
I agree with Sean, it looks like M30ML is a good choice moving forward.

**Reply by capsulecorplab** [reply]
As per July 28th FSW meeting, we'll downselect m30ml for architecture modeling needs for now. Will likely want to visit SysML v2 for a future release
